test/modules/run.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var fs = require('fs');
+var test = require('tape');
+
+require('../../src/ImageSequencer.js');
+
+var sequencer = ImageSequencer({ ui: false });
+var red = "data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAQABADASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAUEAEAAAAAAAAAAAAAAAAAAAAA/8QAFQEBAQAAAAAAAAAAAAAAAAAABgj/xAAUEQEAAAAAAAAAAAAAAAAAAAAA/9oADAMBAAIRAxEAPwCdABykX//Z";
+
+sequencer.loadImages('image1', red);
+sequencer.addSteps('channel');
+sequencer.addSteps('invert');
+sequencer.addSteps('invert');
+
+test('run() works with all possible argument combinations',function(t){
+    sequencer.run();
+    sequencer.run({index: 0});
+    sequencer.run({mode:test},function callback(out){
```

Comment:
looks like `test` should be in quotes? Or do you pass in the test object? 

Also, do you want to somehow try to confirm which steps are actually run? Maybe you could start by running only the last one, and assert that the earlier steps have no output. Then `run()` and see that all have outputs?

Reply:
oh, yeah I'll do that!!